apps/asap-server/test/utils/sync-google-event.test.ts

```diff
@@ -69,56 +110,53 @@ describe('Sync calendar util hook', () => {
       defaultCalendarTimezone,
     );
 
-    expect(eventControllerMock.upsert).toHaveBeenCalledTimes(1);
-    expect(eventControllerMock.upsert).toHaveBeenCalledWith(
-      '04rteq6hj3gfq9g3i8v2oqetvd',
-      {
-        title: 'Event Title',
-        description: 'Event Description',
-        startDate: '2021-02-27T00:00:00.000Z',
-        startDateTimeZone: 'Europe/Lisbon',
-        endDate: '2040-09-13T17:30:00.000Z',
-        endDateTimeZone: 'America/New_York',
-        status: 'Confirmed',
-        calendar: ['squidex-calendar-id'],
-        tags: [],
-      },
-    );
+    expect(eventControllerMock.update).not.toHaveBeenCalled();
+    expect(eventControllerMock.create).toHaveBeenCalledTimes(1);
+    expect(eventControllerMock.create).toHaveBeenCalledWith({
+      googleId: '04rteq6hj3gfq9g3i8v2oqetvd',
+      title: 'Event Title',
+      description: 'Event Description',
+      startDate: '2021-02-27T00:00:00.000Z',
+      startDateTimeZone: 'Europe/Lisbon',
+      endDate: '2040-09-13T17:30:00.000Z',
+      endDateTimeZone: 'America/New_York',
+      status: 'Confirmed',
+      calendar: ['squidex-calendar-id'],
+      tags: [],
+    });
   });
 
   describe('Validation test', () => {
-    test('Should not upsert when validation fails - empty object', async () => {
-      await syncEvent({}, defaultCalendarTimezone);
-      expect(eventControllerMock.upsert).not.toHaveBeenCalled();
+    test('Should reject when validation fails - empty object', async () => {
+      await expect(syncEvent({}, defaultCalendarTimezone)).rejects.toThrow();
     });
 
-    test('Should not upsert when validation fails - missing fields', async () => {
-      await syncEvent({ ...event, id: undefined }, defaultCalendarTimezone);
-      await syncEvent(
-        { ...event, summary: undefined },
-        defaultCalendarTimezone,
-      );
-      expect(eventControllerMock.upsert).not.toHaveBeenCalled();
+    test('Should reject when validation fails - missing fields', async () => {
+      await expect(
+        syncEvent({ ...event, id: undefined }, defaultCalendarTimezone),
+      ).rejects.toThrow();
+      await expect(
```

Comment:
same as previously, we probably shouldn't have multiple invocations of that function and multiple assertions all in one test. 

I do realise how repetitive that might be, you may want to use `test.each` instead as part of the test will always be the same so its really just about testing different input